packages/amplify-cli-logger/src/AmplifyLogger.ts

```diff
@@ -0,0 +1,87 @@
+import winston, { Logger, format } from 'winston';
+import winstonDailyRotateFile from 'winston-daily-rotate-file';
+import { constants } from './constants';
+import { IAmplifyLogger } from './IAmplifyLogger';
+import { JSONUtilities } from 'amplify-cli-core';
+import { getLogFilePath, getLocalLogFilePath, getLogAuditFilePath, getLocalAuditLogFile } from './getLogFilePath';
+import { LocalProjectData, LogPayload, LogErrorPayload } from './Types';
+import { Redactor } from './Redactor';
+
+export class AmplifyLogger implements IAmplifyLogger {
+  logger: Logger;
+  format: winston.Logform.Format;
+  localProjectData!: LocalProjectData;
+  disabledAmplifyLogging: boolean = !!process.env.AMPLIFY_CLI_DISABLE_LOGGING;
+  options: any = {
+    mode: 0o640,
+  };
+  constructor() {
+    this.logger = winston.createLogger();
+    this.format = format.combine(format.timestamp(), format.splat(), format.printf(this.formatter));
+    if (!this.disabledAmplifyLogging) {
+      this.logger.add(
+        new winstonDailyRotateFile({
+          auditFile: getLogAuditFilePath(),
+          filename: getLogFilePath(),
+          datePattern: constants.DATE_PATTERN,
+          maxFiles: `${constants.MAX_FILE_DAYS}d`,
+          handleExceptions: false,
+          format: this.format,
+          //options: this.options,
```

Comment:
Piggybacking on last comment, if we're not including this then we should delete both this comment as well as the config section above.